src/rust/engine/process_execution/src/local.rs

```diff
@@ -163,13 +163,24 @@ pub async fn collect_child_outputs<'a, 'b>(
     stdout: &'a mut BytesMut,
     stderr: &'a mut BytesMut,
     mut stream: BoxStream<'b, Result<ChildOutput, String>>,
+    mut workunit: Option<&mut RunningWorkunit>,
 ) -> Result<i32, String> {
     let mut exit_code = 1;
 
     while let Some(child_output_res) = stream.next().await {
         match child_output_res? {
-            ChildOutput::Stdout(bytes) => stdout.extend_from_slice(&bytes),
-            ChildOutput::Stderr(bytes) => stderr.extend_from_slice(&bytes),
+            ChildOutput::Stdout(bytes) => {
+                stdout.extend_from_slice(&bytes);
+                if let Some(ref mut workunit) = workunit {
+                    workunit.log(bytes.to_vec());
```

Comment:
It looks like this is the only callers of this `log` function (and seemingly transitively `add_log_data` etc.). Did you consider whether we could pass a `Bytes` or similar type, and potentially save the new allocation + O(number of bytes) conversion to `Vec` here?